java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/aggregation/AggFunc.java

```diff
@@ -41,9 +40,11 @@ public interface AggFunc<IN_T, OUT_T, ACC_T> extends Serializable {
     /**
      * Retracts the given input value from the given accumulator.
      *
+     * @param accumulator The accumulator to retract the value from.
      * @param value The value to be retracted.
+     * @param timestamp The timestamp of the value to be retracted.
      */
-    void retract(ACC_T accumulator, IN_T value);
+    void retract(ACC_T accumulator, IN_T value, long timestamp);
```

Comment:
Based on the offline discussion, it would be simpler not to introduce this timestamp field.